sdk/go/common/workspace/project.go

```diff
@@ -87,8 +87,12 @@ type Project struct {
 	// License is the optional license governing this project's usage.
 	License *string `json:"license,omitempty" yaml:"license,omitempty"`
 
-	// Config indicates where to store the Pulumi.<stack-name>.yaml files, combined with the folder Pulumi.yaml is in.
-	Config string `json:"config,omitempty" yaml:"config,omitempty"`
+	// Config has been renamed to StacksDirectory.
+	Config interface{} `json:"config,omitempty" yaml:"config,omitempty"`
+
+	// StacksDirectory indicates where to store the Pulumi.<stack-name>.yaml files, combined with the folder
+	// Pulumi.yaml is in.
+	StacksDirectory string `json:"stacksDirectory,omitempty" yaml:"stacksDirectory,omitempty"`
```

Comment:
Yes, I want to get a few OKs on this before committing to this name.

Reply:
`StackConfigFilesPath`?

Reply:
Thinking off Josh's comment, maybe this should be "ConfigDirectory"? It used to just be called Config, how much do we need to make it clear its for stacks config? 
Other ideas, "Stack[s]ConfigDirectory". "Dir" or "FilesPath" instead of "Directory"?

Reply:
@Frassle What's the proper term for `Pulumi.yaml` versus `Pulumi.stackname.yaml`?

Reply:
*shrug* "Project settings file" and "stack config file" I think.
@pgavlin or @lukehoban might have a definitive answer to that.

Reply:
In our docs - these are currently  "project file" and "stack configuration file".  (Though [this page](https://www.pulumi.com/docs/reference/pulumi-yaml/) includes a weird title that I feel like we should change "Project Configuration Reference").

From that page - the description of the current option is:

> `config`: (optional) directory to store stack-specific configuration files, relative to location of `Pulumi.yaml`.

Here's suggestions in my order of preference:
* `stackConfigDir`
* `stackDir`
* `stackConfigPath`
* `stackPath`
* `stacksDir`
* `stacksPath`

Reply:
stackConfigDir seems fine to me, will update.